language/tools/move-cli/src/main.rs

```diff
@@ -101,33 +96,46 @@ fn maybe_create_dir(dir_name: &str) -> Result<&Path> {
     Ok(dir)
 }
 
-/// Compile the user modules in `move_src_directory` and the script in `script_file`
+/// Compile the user modules in `move_src` and the script in `script_file`
 fn compile(
     args: &Move,
     script_file: &Option<String>,
 ) -> Result<(OnDiskStateView, Option<CompiledScript>)> {
-    let move_src = maybe_create_dir(&args.move_src)?;
+    let move_src = Path::new(&args.move_src);
     let move_data = maybe_create_dir(&args.move_data)?;
 
     // allow user modules in move_src directory to compile against stdlib and move_data modules, but
     // not Libra framework modules
     // TODO: prevent cyclic dep?
-    let mut user_lib_files: Vec<String> = fs::read_dir(move_src)?
-        .map(|f| f.map_or("".to_string(), |f| f.path().to_string_lossy().into_owned()))
-        .filter(|p| p.ends_with(".move"))
-        .collect();
+    let mut user_lib_files: Vec<String> = if move_src.exists() {
+        fs::read_dir(move_src)?
+            .map(|f| f.map_or("".to_string(), |f| f.path().to_string_lossy().into_owned()))
+            .filter(|p| p.ends_with(".move"))
+            .collect()
+    } else {
+        vec![]
+    };
 
-    if !user_lib_files.is_empty() {
-        println!("Compiling {:?} user module(s)", user_lib_files.len());
+    let has_user_modules = !user_lib_files.is_empty();
+
+    if has_user_modules {
+        print!("Compiling {:?} module(s)", user_lib_files.len());
```

Comment:
Fine for now either way, but should this live behind a --verbose flag?

Reply:
Yeah, I have it on my plate to add a `verbose` or `quiet` flag. The printing is already a bit out of hand.